src/modules/paging/all.js

```diff
@@ -0,0 +1,30 @@
+import { flatten, range } from 'lodash';
+import { MaxLimitReached } from '../../error';
+import concurrentPaginate from '../concurrency';
+
+export default async function all(
+    req,
+    { searchRequest, args = {}, maxLimit = null, concurrencyLimit = 1 },
+) {
+    const { limit, offset, ...searchArgs } = args;
+
+    const params = {
+        ...searchArgs,
+        limit: limit || 20,
+        offset: offset || 0,
+    };
+    const firstPage = await searchRequest(req, params);
+    if (firstPage.offset + firstPage.limit >= firstPage.count) {
```

Comment:
it can be really nice to handle `undefined` offset and limit results - for unitests purposes. It will allow us to be lazy.